appstudio-controller/controllers/appstudio.redhat.com/devsandbox_deployment_controller.go

```diff
@@ -0,0 +1,225 @@
+		// Don't requeue if the requested object is not found/deleted.
+		if apierr.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	dt, err := findMatchingDTForSpaceRequest(ctx, r.Client, &spacerequest)
+	if err != nil {
+		log.Error(err, "error while finding a dt that matches the SpaceRequest")
+		return ctrl.Result{}, err
+	}
+
+	if dt == nil {
+		dt, err = CreateDeploymentTargetForSpaceRequest(ctx, r.Client, &spacerequest)
+		if err != nil {
+			log.Error(err, "failed to create the DeploymentTarget for a SpaceRequest")
+			return ctrl.Result{}, err
+		}
+
+		log.Info("DeploymentTarget is created for SpaceRequest", "SpaceRequest.Name", spacerequest.Name,
+			"SpaceRequest.Namespace", spacerequest.Namespace)
+	}
+
+	log.Info("A DeploymentTarget for the SpaceRequest exists", "DeploymentTarget.Name", dt.Name, "Namespace", dt.Namespace)
+
+	err = updateDTStatusPhase(ctx, r.Client, dt, applicationv1alpha1.DeploymentTargetPhase_Available, log)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// findMatchingDTForSpaceRequest tries to find a DT that matches the given SpaceRequest in a namespace.
+func findMatchingDTForSpaceRequest(ctx context.Context, k8sClient client.Client, spacerequest *codereadytoolchainv1alpha1.SpaceRequest) (*applicationv1alpha1.DeploymentTarget, error) {
+	if dtc, _ := findMatchingDTCForSpaceRequest(ctx, k8sClient, spacerequest); dtc != nil {
+		dtList := applicationv1alpha1.DeploymentTargetList{}
+		opts := []client.ListOption{
+			client.InNamespace(spacerequest.Namespace),
+		}
+
+		if err := k8sClient.List(ctx, &dtList, opts...); err != nil {
+			return nil, err
+		}
+
+		var dt *applicationv1alpha1.DeploymentTarget
+		for i, d := range dtList.Items {
+			if dtc.Name == string(d.Spec.ClaimRef) {
+				dt = &dtList.Items[i]
+				return dt, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func HasLabel(object client.Object, label string) bool {
+	_, found := object.GetLabels()[label]
+	return found
+}
+
+func HasAnnotation(object client.Object, annotation string) bool {
+	_, found := object.GetAnnotations()[annotation]
+	return found
+}
+
+// findMatchingDTCForSpaceRequest tries to find the DTC that matches a given SpaceRequest.
+func findMatchingDTCForSpaceRequest(ctx context.Context, k8sClient client.Client, spacerequest *codereadytoolchainv1alpha1.SpaceRequest) (*applicationv1alpha1.DeploymentTargetClaim, error) {
+	if !HasLabel(spacerequest, deploymentTargetClaimLabel) {
+		return nil, fmt.Errorf("no 'appstudio.openshift.io/dtc' label is set for spacerequest %s %s", spacerequest.Name, spacerequest.Namespace)
+	}
+	dtcList := applicationv1alpha1.DeploymentTargetClaimList{}
+	opts := []client.ListOption{
+		client.InNamespace(spacerequest.Namespace),
+	}
+
+	if err := k8sClient.List(ctx, &dtcList, opts...); err != nil {
+		return nil, err
+	}
+
+	if len(dtcList.Items) > 0 {
```

Comment:
It might be more performant to add the `client.MatchingLabels{deploymentTargetClaimLabel: dtc.Name,}` to the `opts` on line 146, wdyt?

Reply:
The  deploymentTargetClaimLabel label is in SpaceRequest CRs. Here is to find the DeploymentTargetClaim CRs.  So, it's not used.

Reply:
Ah, I read the code the other way around, now I understand, thanks. 
Maybe add a comment above this line explaining the logic for future reference?

Reply:
Updated as you suggested. Thanks

Reply:
Thanks!